File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0, n = numbers.length - 1;

        while(l < n) {
            int curSum = numbers[l] + numbers[n];
            if(curSum > target) {
                n--;
            } else if(curSum < target) {
                l++;
            } else {
                return new int[] {l + 1, n + 1};
            }
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/combination-target-sum-ii/submission-2.py
class Solution:
    def combinationSum2(self, candidates: List[int], target: int) -> List[List[int]]:
        res, comb = [], []
        candidates.sort()

        def backtrack(start, total):
            if total == target:
                res.append(comb[:])
                return

            if total > target:
                return
            
            for i in range(start, len(candidates)):
                if i > start and candidates[i] == candidates[i - 1]:
                    continue
                comb.append(candidates[i])
                backtrack(i + 1, total + candidates[i])
                comb.pop()
        backtrack(0, 0)
        return res